starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecBridge.java

```diff
@@ -957,45 +957,43 @@ private int queueSecureInputBuffer(
       int[] numBytesOfEncryptedData,
       int numSubSamples,
       int cipherMode,
-      int blocksToEncrypt,
-      int blocksToSkip,
+      int patternEncrypt,
+      int patternSkip,
       long presentationTimeUs) {
     resetLastPresentationTimeIfNeeded(presentationTimeUs);
     try {
+      boolean usesCbcs = cipherMode == MediaCodec.CRYPTO_MODE_AES_CBC;
+      if (usesCbcs && !MediaDrmBridge.isCbcsSchemeSupported()) {
+        Log.e(TAG, "Encryption scheme 'cbcs' not supported on this platform.");
+        return MediaCodecStatus.ERROR;
+      }
       CryptoInfo cryptoInfo = new CryptoInfo();
       cryptoInfo.set(
           numSubSamples, numBytesOfClearData, numBytesOfEncryptedData, keyId, iv, cipherMode);
-
-      if (cipherMode == MediaCodec.CRYPTO_MODE_AES_CBC) {
-        cryptoInfo.setPattern(new Pattern(blocksToEncrypt, blocksToSkip));
-      } else if (blocksToEncrypt != 0 || blocksToSkip != 0) {
-        Log.e(TAG, "Pattern encryption only supported for 'cbcs' scheme (CBC mode).");
-        return MediaCodecStatus.ERROR;
+      if (patternEncrypt != 0 && patternSkip != 0) {
+        if (usesCbcs) {
+          // Above platform check ensured that setting the pattern is indeed supported.
+          cryptoInfo.setPattern(new Pattern(patternEncrypt, patternSkip));
+        } else {
+          Log.e(TAG, "Pattern encryption only supported for 'cbcs' scheme (CBC mode).");
+          return MediaCodecStatus.ERROR;
+        }
       }
-
       mMediaCodec.get().queueSecureInputBuffer(index, offset, cryptoInfo, presentationTimeUs, 0);
     } catch (MediaCodec.CryptoException e) {
-      int errorCode = e.getErrorCode();
-      if (errorCode == MediaCodec.CryptoException.ERROR_NO_KEY) {
+      if (e.getErrorCode() == MediaCodec.CryptoException.ERROR_NO_KEY) {
         Log.d(TAG, "Failed to queue secure input buffer: CryptoException.ERROR_NO_KEY");
         return MediaCodecStatus.NO_KEY;
-      } else if (errorCode == MediaCodec.CryptoException.ERROR_INSUFFICIENT_OUTPUT_PROTECTION) {
-        Log.d(
-            TAG,
-            "Failed to queue secure input buffer: "
-                + "CryptoException.ERROR_INSUFFICIENT_OUTPUT_PROTECTION");
-        return MediaCodecStatus.INSUFFICIENT_OUTPUT_PROTECTION;
```

Comment:
Consider keeping the `INSUFFICIENT_OUTPUT_PROTECTION` path.

Reply:
Done